rp2040-hal/src/xosc.rs

```diff
@@ -184,6 +181,7 @@ impl CrystalOscillator<Stable> {
             w
         });
 
-        self.transition(Dormant)
+        let freq_hz = self.state.freq_hz;
+        self.transition(Unstable { freq_hz })
```

Comment:
In case the system clock is currently running from the ROSC, transitioning XOSC to dormant will not stop the processor. Am I missing something, or would this function just return `CrystalOscillator<Unstable>` even though the oscillator is dormant?

I'm not sure if there are valid use cases for calling this function in such a way. If not, perhaps just mention this behavior in the safety comment? ("This should only be called if this oscillator is the clock source for the system clock" or similar)?

Reply:
Yes, good point @jannic. That's my understanding too.

I've added comments to that effect for the rosc and xosc `dormant()` methods.

Reply:
Great!
If it turns out that switching the non-active oscillator to dormant is needed for some use case, we probably need to bring the `Dormant` state back. But I'm fine with merging this as it is. In any useful configuration I see at the moment, the other oscillator can just be disabled, like you do in the example.

Reply:
That's right, I didn't think about this case 🤔.
When applied to the core's source, the state transitions visible from the core are: `Running -> Unstable -> Stable`
And from the other source it's `Running -> Dormant -> Unstable -> Stable`.

Would it make sense to have something like:
```rust
fn dormant(self) -> Either<CrystalOscillator<Unstable>, CrystalOscillator<Dormant>>
```

Reply:
Yes, such a return type would be exactly what I'd suggest if we want to support setting the non-active oscillator to dormant.
As it would require additional code to determine the actual mode, and complicates the return type, I hesitate to add it just for completeness. That's why I think about actual use cases.

Reply:
Yes, I like the idea of keeping the implementation as ergonomic as possible for likely use cases.